### PR TITLE
[dynamo] Remove SuperSource

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110375
* #110535
* __->__ #110475

The motivation for removing this is already present in the pre-PR comments. Copying it

~~~
# NB - SuperSource is a weird one.
# it is our only source with 2 bases, so we use the objec
# as the base, rather than the type, since an invocation
# like super(Foo, foo) is represented here, the source object base is more spiritually
# aligned with the instance, rather than the type.
# This whole construction is questionable tho, and we should probably find a way to
# avoid this exception to our otherwise nice source parentage invariant.
~~~

Instead of using super(a, b), we can use `type(b).__mro__[index]`.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/variables/misc.py [hunk 1]
```diff
@@ -552,70 +639,8 @@ def reconstruct(self, codegen):
     def call_function(
         self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
     ) -> "VariableTracker":
-        from .builder import wrap_fx_proxy
```
Comment (anijain2305) [inline torch/_dynamo/variables/misc.py:555]: Code motion - Moved to the call_method of SuperVariable. That is a better place for this code.

File: torch/_dynamo/variables/misc.py [hunk 2]
```diff
@@ -552,70 +639,8 @@ def reconstruct(self, codegen):
...
-                        **options,
-                    )
-                else:
-                    unimplemented(
-                        f"GetAttrVariable.call_function original __torch_function__ {args}"
-                    )
-
         if isinstance(self.obj, AutogradFunctionVariable) and self.name == "apply":
             return self.obj.call_apply(tx, args, kwargs).add_options(self)
-        # calling parent class‘s non classmethod from child class
```
Comment (anijain2305) [inline torch/_dynamo/variables/misc.py:609]: Does not seem necessary. The tests added for this fix go through a different codepath.

**Comment by anijain2305** [comment]
@jansel @voznesenskym @yanboliang Gentle ping.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**